polyfills/Array/prototype/lastIndexOf/polyfill.js

```diff
@@ -1,20 +1,42 @@
-Array.prototype.lastIndexOf = function lastIndexOf(searchElement) {
-	if (this === undefined || this === null) {
-		throw new TypeError(this + ' is not an object');
+// 22.1.3.15. Array.prototype.lastIndexOf ( searchElement [ , fromIndex ] )
+CreateMethodProperty(Array.prototype, 'lastIndexOf', function lastIndexOf(searchElement /* [ , fromIndex ] */) {
+	// 1. Let O be ? ToObject(this value).
+	var O = ToObject(this);
+	// 2. Let len be ? ToLength(? Get(O, "length")).
+	var len = ToLength(Get(O, "length"));
+	// 3. If len is 0, return -1.
+	if (len === 0) {
+		return -1;
 	}
-
-	var
-	arraylike = this instanceof String ? this.split('') : this,
-	length = Math.max(Math.min(arraylike.length, 9007199254740991), 0) || 0,
-	index = Number(arguments[1]) || 0;
-
-	index = 1 in arguments ? (index < 0 ? Math.max(length + index, 0) : index) + 1 : length;
-
-	while (--index >= 0) {
-		if (index in arraylike && arraylike[index] === searchElement) {
-			return index;
+	// 4. If fromIndex is present, let n be ? ToInteger(fromIndex); else let n be len-1.
+	var n = arguments.length > 1 ? arguments[1] : len - 1;
```

Comment:
Why no `ToInteger` here?

Reply:
Something I missed, good catch, I'll add it now 👍